backend/services/core/migrations/20230822094308_add_custom_author_type_to_articles.ts

```diff
@@ -0,0 +1,106 @@
+import { Knex } from 'knex';
+
+export async function up(knex: Knex): Promise<void> {
+  // Add custom authors table
+  await knex.schema.createTable('custom_authors', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'));
+    table.string('name').notNullable().comment('Swedish name of custom author');
+    table.string('name_en').comment('English name of custom author, possibly null');
+    table.string('image_url').comment('Image url of custom author, possibly null');
+    table.timestamps(true, true);
+  });
+  await knex.schema.createTable('custom_author_roles', (table) => {
+    table.uuid('id').primary().defaultTo(knex.raw('gen_random_uuid()'));
+    table.uuid('custom_author_id').unsigned().notNullable().references('custom_authors.id')
+      .onDelete('CASCADE');
+    table.string('role').unsigned().notNullable().comment('should use role syntax, such as dsek.styr or dsek.stab.noll');
+    table.timestamps(true, true);
+  });
+  // Add new "authors" table
+  await knex.schema.createTable('authors', (table) => {
```

Comment:
I think this is an easy fix. We could remove the type column in its entirety and only ever have the priority CustomAuthor > Mandate > Member, with null checks to determine the type of author.

For enforce database consistency we could add a check constraint to only ever allow one of the two columns mandate_id and custom_author_id to be non-null, not both at the same time. https://knexjs.org/guide/schema-builder.html#checks

Reply:
The SO post I linked includes a suggestion to have FK in the opposite direction (on the 'child' tables instead of on author) which yields a clean authors table, but I don't really understand how it's supposed to work in our case (would require primary key spanning more than one table). 

CHECK constraints are an easy solution, but it's still bothers me a bit that there's a implicit priority to the types of authors.

Reply:
Couldn't we keep type and have check constraints that is in pseudo-logic:
(Type = 'Mandate' && mandate_id IS NOT NULL && custom_author_id IS NULL)
OR (...)
OR (...)
I think you understand the ...s.